libraries/adaptive-expressions/src/functionUtils.ts

```diff
@@ -0,0 +1,1045 @@
+     */
+    public static verifyNumber(value: any, expression: Expression, _: number): string | undefined {
+        let error: string;
+        if (!FunctionUtils.isNumber(value)) {
+            error = `${expression} is not a number.`;
+        }
+
+        return error;
+    }
+
+    /**
+     * Verify value is numeric.
+     * @param value value to check.
+     * @param expression Expression that led to value.
+     * @returns Error or undefined if invalid.
+     */
+    public static verifyNumberOrNumericList(value: any, expression: Expression, _: number): string | undefined {
+        let error: string;
+        if (FunctionUtils.isNumber(value)) {
+            return error;
+        }
+
+        if (!Array.isArray(value)) {
+            error = `${expression} is neither a list nor a number.`;
+        } else {
+            for (const elt of value) {
+                if (!FunctionUtils.isNumber(elt)) {
+                    error = `${elt} is not a number in ${expression}.`;
+                    break;
+                }
+            }
+        }
+
+        return error;
+    }
+
+    /**
+     * Verify value is numeric list.
+     * @param value value to check.
+     * @param expression Expression that led to value.
+     * @returns Error or undefined if invalid.
+     */
+    public static verifyNumericList(value: any, expression: Expression, _: number): string | undefined {
+        let error: string;
+        if (!Array.isArray(value)) {
+            error = `${expression} is not a list.`;
+        } else {
+            for (const elt of value) {
+                if (!FunctionUtils.isNumber(elt)) {
+                    error = `${elt} is not a number in ${expression}.`;
+                    break;
+                }
+            }
+        }
+
+        return error;
+    }
+
+    /**
+     * Verify value contains elements.
+     * @param value value to check.
+     * @param expression Expression that led to value.
+     * @returns Error or undefined if invalid.
+     */
+    public static verifyContainer(value: any, expression: Expression, _: number): string | undefined {
+        let error: string;
+        if (!(typeof value === 'string') && !Array.isArray(value) && !(value instanceof Map) && !(typeof value === 'object')) {
+            error = `${expression} must be a string, list, map or object.`;
+        }
+
+        return error;
+    }
+
+    /**
+     * Verify value is not null or undefined.
+     * @param value value to check.
+     * @param expression Expression that led to value.
+     * @returns Error or undefined if valid.
+     */
+    public static verifyNotNull(value: any, expression: Expression, _: number): string | undefined {
+        let error: string;
+        if (value === undefined || value === null) {
+            error = `${expression} is null.`;
+        }
+
+        return error;
+    }
+
+    /**
+     * Verify value is an integer.
+     * @param value value to check.
```

Comment:
>value value [](start = 14, length = 11)

You should do a case sensitive search and replace on "value value" to "value Value"